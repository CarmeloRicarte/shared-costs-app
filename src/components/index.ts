export * from './CustomTable/CustomTable';
export * from './Modal';
