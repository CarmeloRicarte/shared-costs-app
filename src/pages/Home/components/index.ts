export * from './Balance';
export * from './CostsList';
export * from './CostsListItem';
