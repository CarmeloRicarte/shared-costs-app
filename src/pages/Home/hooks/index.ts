export * from './useBalances';
export * from './useCosts';
export * from './useFriends';
export * from './useGroups';
