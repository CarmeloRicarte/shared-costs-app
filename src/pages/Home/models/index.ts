export * from './Balance';
export * from './Cost';
export * from './Friend';
export * from './Group';
