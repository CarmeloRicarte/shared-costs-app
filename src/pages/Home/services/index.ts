export * from './Costs.service';
export * from './Friends.service';
export * from './Groups.service';
