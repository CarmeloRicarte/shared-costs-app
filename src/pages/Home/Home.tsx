import { useEffect, useState } from 'react';

import { Modal } from '../../components';
import { LoadingSpinner } from '../../components/LoadingSpinner';
import { AddFriendForm, AddPaymentForm, BalanceByFriend } from './components';
import { CostsList } from './components/CostsList';
import { useCosts } from './hooks';

export const Home = () => {
  const [isModalAddFriendOpen, setIsModalAddFriendOpen] = useState(false);
  const [isModalAddPaymentOpen, setIsModalAddPaymentOpen] = useState(false);
  const { costs, getAllCosts, isLoading } = useCosts();

  const handleToggleAddFriendModal = () => {
    setIsModalAddFriendOpen(!isModalAddFriendOpen);
  };

  const handleToggleAddPaymentModal = () => {
    setIsModalAddPaymentOpen(!isModalAddPaymentOpen);
  };

  const getCostsData = async () => {
    try {
      await getAllCosts();
    } catch (error) {
      console.error(error);
    }
  };

  useEffect(() => {
    void getCostsData();
  }, []);

  return (
    <>
      <h1 className='text-center'>Gastos Compartidos</h1>
      <div className='list-grid-2-columns'>
        {isLoading ? (
          <LoadingSpinner />
        ) : (
          <>
            <CostsList
              costs={costs}
              onClickPagoButton={handleToggleAddPaymentModal}
              onClickAmigoButton={handleToggleAddFriendModal}
            />
            <BalanceByFriend />
          </>
        )}
      </div>
      <Modal
        isOpen={isModalAddFriendOpen}
        onClose={handleToggleAddFriendModal}
        headerTitle='Añadir amigo al grupo'
      >
        <AddFriendForm onSubmitForm={handleToggleAddFriendModal} />
      </Modal>
      <Modal
        isOpen={isModalAddPaymentOpen}
        onClose={handleToggleAddPaymentModal}
        headerTitle='Añadir pago'
      >
        <AddPaymentForm onSubmitForm={handleToggleAddPaymentModal} />
      </Modal>
    </>
  );
};
