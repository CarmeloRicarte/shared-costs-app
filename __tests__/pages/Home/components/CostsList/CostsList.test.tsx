import { fireEvent, render, screen } from '@testing-library/react';
import { CostList } from 'pages/Home/components';

import { mockCosts } from '../../../../__fixtures__';

describe('CostsList tests', () => {
  const onClickAmigoButtonMock = vi.fn();
  const onClickPagoButtonMock = vi.fn();

  test('should render the costs list', () => {
    const { container } = render(
      <CostList
        costs={mockCosts}
        onClickAmigoButton={onClickAmigoButtonMock}
        onClickPagoButton={onClickPagoButtonMock}
      />
    );

    expect(container).toMatchSnapshot();
  });

  test('should call to onClickPagoButton on click button "A単adir pago"', () => {
    render(
      <CostList
        costs={mockCosts}
        onClickAmigoButton={onClickAmigoButtonMock}
        onClickPagoButton={onClickPagoButtonMock}
      />
    );

    const button = screen.getByRole('button', { name: 'A単adir pago' });
    fireEvent.click(button);
    expect(onClickPagoButtonMock).toHaveBeenCalled();
  });

  test('should call to onClickAmigoButton on click button "A単adir amigo"', () => {
    render(
      <CostList
        costs={mockCosts}
        onClickAmigoButton={onClickAmigoButtonMock}
        onClickPagoButton={onClickPagoButtonMock}
      />
    );

    const button = screen.getByRole('button', { name: 'A単adir amigo' });
    fireEvent.click(button);
    expect(onClickAmigoButtonMock).toHaveBeenCalled();
  });
});
