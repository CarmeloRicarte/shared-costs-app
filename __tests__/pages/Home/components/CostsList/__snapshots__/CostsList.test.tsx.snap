// Vitest Snapshot v1

exports[`CostsList tests > should render the costs list 1`] = `
<div>
  <section
    class="default-section"
  >
    <ul
      class="costs-list"
    >
      <div
        class="costs-list-grid__header"
      >
        <h2>
          Listado de gastos
        </h2>
        <button
          class="action-button"
        >
          Añadir pago
        </button>
        <button
          class="action-button"
        >
          Añadir amigo
        </button>
      </div>
      <li>
        <div
          class="costs-list-grid__item"
        >
          <div>
            <p
              class="costs-list-grid__personName"
            >
              Pedro
            </p>
            <p
              class="costs-list-grid__description"
            >
              Comida
            </p>
          </div>
          <div
            class="text-end"
          >
            <p
              class="costs-list-grid__totalAmount"
            >
              100
               €
            </p>
            <p
              class="costs-list-grid__paymentDate"
            >
              18/2/2023, 22:00:00
            </p>
          </div>
        </div>
      </li>
      <li>
        <div
          class="costs-list-grid__item"
        >
          <div>
            <p
              class="costs-list-grid__personName"
            >
              Juan
            </p>
            <p
              class="costs-list-grid__description"
            >
              Gasolina
            </p>
          </div>
          <div
            class="text-end"
          >
            <p
              class="costs-list-grid__totalAmount"
            >
              50
               €
            </p>
            <p
              class="costs-list-grid__paymentDate"
            >
              17/2/2023, 21:00:00
            </p>
          </div>
        </div>
      </li>
    </ul>
  </section>
</div>
`;
